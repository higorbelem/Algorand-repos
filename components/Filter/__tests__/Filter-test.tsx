import * as React from 'react';
import { render } from '@testing-library/react-native';

import { Filter } from '../index';

const mockData = [{id: 'perawallet', label: 'Pera Wallet'}, {id: 'algorandfoundation', label: 'Algorand Foundation'}, {id: 'algorand', label: 'Algorand'}];

describe('Filter component', () => {
  it('Renders correctly', () => {
    const tree = render(<Filter data={mockData} onItemSelected={() => {}}/>).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Show correct number of items', async () => {
    const tree = render(<Filter data={mockData} onItemSelected={() => {}}/>);

    const components = await tree.findAllByTestId('filter-item');

    expect(components).toHaveLength(mockData.length + 1);
  });

  it('Show "All" item', async () => {
    const tree = render(<Filter data={mockData} onItemSelected={() => {}}/>);

    const component = tree.getByText('All');

    expect(component).toBeTruthy();
  });
});