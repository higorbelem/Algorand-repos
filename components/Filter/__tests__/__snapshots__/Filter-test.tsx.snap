// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component Renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "alignItems": "center",
      "gap": 10,
      "justifyContent": "space-between",
      "paddingHorizontal": 20,
    }
  }
  horizontal={true}
  showsHorizontalScrollIndicator={false}
  style={
    {
      "flexGrow": 0,
    }
  }
>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#2D2DF1",
          "borderRadius": 20,
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "paddingHorizontal": 20,
        }
      }
      testID="filter-item"
    >
      <Text
        style={
          [
            {
              "color": "#FFFFFF",
            },
            undefined,
            {
              "fontSize": 16,
              "fontWeight": "400",
            },
          ]
        }
      >
        All
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#ffffff22",
          "borderRadius": 20,
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "paddingHorizontal": 20,
        }
      }
      testID="filter-item"
    >
      <Text
        style={
          [
            {
              "color": "#FFFFFF",
            },
            undefined,
            {
              "fontSize": 16,
              "fontWeight": "400",
            },
          ]
        }
      >
        Pera Wallet
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#ffffff22",
          "borderRadius": 20,
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "paddingHorizontal": 20,
        }
      }
      testID="filter-item"
    >
      <Text
        style={
          [
            {
              "color": "#FFFFFF",
            },
            undefined,
            {
              "fontSize": 16,
              "fontWeight": "400",
            },
          ]
        }
      >
        Algorand Foundation
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#ffffff22",
          "borderRadius": 20,
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "paddingHorizontal": 20,
        }
      }
      testID="filter-item"
    >
      <Text
        style={
          [
            {
              "color": "#FFFFFF",
            },
            undefined,
            {
              "fontSize": 16,
              "fontWeight": "400",
            },
          ]
        }
      >
        Algorand
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
